x-pack/plugins/security_solution/server/lib/detection_engine/rule_management/normalization/rule_converters.ts

```diff
@@ -437,15 +446,35 @@ export const patchTypeSpecificSnakeToCamel = (
 interface ConvertUpdateAPIToInternalSchemaProps {
   existingRule: SanitizedRule<RuleParams>;
   ruleUpdate: RuleUpdateProps;
+  actionsClient: ActionsClient;
 }
 
 export const convertUpdateAPIToInternalSchema = ({
   existingRule,
   ruleUpdate,
+  actionsClient,
 }: ConvertUpdateAPIToInternalSchemaProps) => {
+  const [ruleUpdateSystemActions, ruleUpdateActions] = partition(ruleUpdate.actions, (action) =>
+    actionsClient.isSystemAction(action.id)
+  );
```

Comment:
Can we factor out this logic to something like `separateActionsAndSystemAction` into a reusable function? Then it can be reused also below for doing the same with existing actions (and also the same logic is used in `migrateLegacyActionsIds` in `x-pack/plugins/security_solution/server/lib/detection_engine/rule_management/utils/utils.ts` )